src/chgrp/chgrp.rs

```diff
@@ -235,7 +235,12 @@ impl Chgrper {
             let may_exist = if follow_arg {
                 path.canonicalize().ok()
             } else {
-                Some(resolve_relative_path(path).into_owned())
+                let real = resolve_relative_path(path);
+                if real.is_dir() {
```

Comment:
@wimh How GNU chgrp treats `/some/path` and `/some/path/` is slightly different. The former is regarded as a file even if it is a symlink to a directory, but the latter is always considered as a directory, in other words, it will be dereferenced.

To achieve  that is a bit tricky in Rust, so I decided to prohibit changing ownership as long as the path is pointing at '/'.